src/Evolution/DiscontinuousGalerkin/LiftFromBoundary.cpp

```diff
@@ -0,0 +1,243 @@
+        const size_t volume_index = si.offset() + si.stride() * index_on_stripe;
+        volume_dt_vars.get()[component_index * num_volume_pts + volume_index] -=
+            volume_det_inv_jacobian[volume_index] *
+            boundary_lifting_term[index_on_stripe] *
+            get(face_det_jacobian)[boundary_index] *
+            get(magnitude_of_face_normal)[boundary_index] *
+            boundary_corrections[component_index * num_boundary_pts +
+                                 boundary_index];
+      }
+    }
+  }
+}
+
+template <size_t Dim>
+void lift_boundary_terms_gauss_points_impl(
+    const gsl::not_null<double*> volume_dt_vars,
+    const size_t num_independent_components, const Mesh<Dim>& volume_mesh,
+    const size_t dimension,
+    const gsl::span<const double>& volume_det_inv_jacobian,
+    const size_t num_boundary_pts,
+    const gsl::span<const double>& upper_boundary_corrections,
+    const DataVector& upper_boundary_lifting_term,
+    const Scalar<DataVector>& upper_magnitude_of_face_normal,
+    const Scalar<DataVector>& upper_face_det_jacobian,
+    const gsl::span<const double>& lower_boundary_corrections,
+    const DataVector& lower_boundary_lifting_term,
+    const Scalar<DataVector>& lower_magnitude_of_face_normal,
+    const Scalar<DataVector>& lower_face_det_jacobian) noexcept {
+  const size_t num_volume_pts = volume_mesh.number_of_grid_points();
+  if (dimension == 0) {
+    lift_boundary_terms_gauss_points_impl_xi_dir(
+        volume_dt_vars, num_independent_components, num_volume_pts,
+        volume_det_inv_jacobian, num_boundary_pts, upper_boundary_corrections,
+        upper_boundary_lifting_term, upper_magnitude_of_face_normal,
+        upper_face_det_jacobian, lower_boundary_corrections,
+        lower_boundary_lifting_term, lower_magnitude_of_face_normal,
+        lower_face_det_jacobian);
+    return;
+  }
+  // Developer note: A potential optimization is to re-order (not transpose!)
+  // the volume time derivative for all variables before lifting, so that the
+  // lifting can be done using vectorized math and DataVectors as views. It
+  // would need to be tested whether that actually increases performance.
+  // Another alternative would be to use a SIMD library, such as nsimd or xsimd.
+
+  const size_t stripe_size = volume_mesh.extents(dimension);
+  size_t boundary_index = 0;
+  for (StripeIterator si{volume_mesh.extents(), dimension}; si;
+       (void)++si, (void)++boundary_index) {
+    // Loop over each stripe in this logical direction. This is effectively
+    // looping over each boundary grid point.
+    for (size_t component_index = 0;
+         component_index < num_independent_components; ++component_index) {
+      for (size_t index_on_stripe = 0; index_on_stripe < stripe_size;
+           ++index_on_stripe) {
+        const size_t volume_index = si.offset() + si.stride() * index_on_stripe;
+        volume_dt_vars.get()[component_index * num_volume_pts + volume_index] -=
```

Comment:
did you consider making the `component_index` as the inner loop? you could then move some multiplications outside of the component loop... (this question also applies to the other impl)

Reply:
ping on if you had any thoughts on this...

Reply:
I did. The reason I went with this is because making the inner loop over the component would mean more jumping around in memory. I'm not sure which would be faster since jumping around in memory is bad for performance.